Automattic/wp-calypso
# Update empty cart redirect to use leaveCheckout

#### Changes proposed in this Pull Request

This PR will update the existing usage of `clickClose` found on the empty cart page's call to action button to use [the new leaveCheckout](https://github.com/Automattic/wp-calypso/pull/59226) function.

Since leaveCheckout had the tracks event hardcoded, a tracksEvent prop is being added in this PR. Because of this, the two existing usages of leaveCheckout needed the tracks event prop passed along as well.

#### Testing instructions


* Enable [tracks event monitoring](PCYsg-cae-p2) in console
* Add a domain to your cart, skip over adding an email when prompted
* In checkout, click the X button in the masterbar to go back to the domains checkout flow (last page you were on, it should be the page that prompts you to add email)
* Check console for `calypso:analytics Recording event "calypso_masterbar_close_clicked" with actual props`
* In the domain's checkout flow, click on Skip in the upper right corner to open checkout again
* Now, empty the cart to trigger a redirection back to the domain's checkout flow
* Now a new tracks event is fired `calypso:analytics Recording event "calypso_empty_cart_redirect" with actual props`
* Lastly, click on Skip again. This will now bring you to the empty cart page because the cart is now empty.
* Click on the Go Back call to action button on the empty cart page.
* This should fire a separate tracks event `calypso:analytics Recording event "calypso_checkout_composite_empty_cart_clicked" with actual props`



Related to 535-gh-Automattic/payments-shilling#issuecomment-988219990
